### PR TITLE
Add Subcription filtering

### PR DESCRIPTION
**Pull request by sirojan24**
Modify Queue and Topic subscription UI and  add filtering service

### CODE REVIEW

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp [hunk 1]
```diff
@@ -78,28 +83,85 @@
...
+            identifierPattern = "";
+        }
+        ClusterManagerClient client;
+        String[] allClusterNodeAddresses;
+        boolean isClusteringEnabled = false;
+        String serverURL = CarbonUIUtil.getServerURL(config.getServletContext(), session);
+        ConfigurationContext configContext = (ConfigurationContext) config.getServletContext().getAttribute
+        (CarbonConstants.CONFIGURATION_CONTEXT);
+        String cookie = (String) session.getAttribute(ServerConstants.ADMIN_SERVICE_COOKIE);
+        String nodeID = "";
```
Comment (hastef88) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp]: should be nodeId.

File: components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/internal/registry/SubscriptionManagementBeans.java [hunk 2]
```diff
@@ -92,6 +92,122 @@ public static SubscriptionManagementBeans getInstance(){
...
+            String[] signature = new String[]{boolean.class.getName(), boolean.class.getName(),
+                    String.class.getName(), String.class.getName(), String.class.getName(), boolean.class.getName(),
+                    String.class.getName(), boolean.class.getName(), String.class.getName(),  int.class.getName(),
+                    int.class.getName()};
+
+            Object result = mBeanServer.invoke(objectName,
+                    SubscriptionManagementConstants.FILTERED_SUBSCRIPTIONS_MBEAN_ATTRIBUTE,
+                    parameters, signature);
+
+            if (result != null) {
```
Comment (sdkottegoda) [inline components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/internal/registry/SubscriptionManagementBeans.java]: Let's flip the null check

File: components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/internal/registry/SubscriptionManagementBeans.java [hunk 3]
```diff
@@ -92,6 +92,122 @@ public static SubscriptionManagementBeans getInstance(){
...
+                                                            String filteredNamePattern,
+                                                            boolean isFilteredNameByExactMatch, String identifierPattern,
+                                                            boolean isIdentifierPatternByExactMatch, String ownNodeId,
+                                                            int pageNumber, int subscriptionCountPerPage)
+                                                            throws SubscriptionManagerException {
+        ArrayList<Subscription> subscriptionDetailsList = new ArrayList<>();
+        MBeanServer mBeanServer = ManagementFactory.getPlatformMBeanServer();
+        try {
+            ObjectName objectName =
+                    new ObjectName("org.wso2.andes:type=SubscriptionManagementInformation," +
```
Comment (sdkottegoda) [inline components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/internal/registry/SubscriptionManagementBeans.java]: + needs to be at the beginning

File: components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/internal/registry/SubscriptionManagementBeans.java [hunk 4]
```diff
@@ -92,6 +92,122 @@ public static SubscriptionManagementBeans getInstance(){
...
+                    isIdentifierPatternByExactMatch, ownNodeId, pageNumber,
+                    subscriptionCountPerPage};
+            String[] signature = new String[]{boolean.class.getName(), boolean.class.getName(),
+                    String.class.getName(), String.class.getName(), String.class.getName(), boolean.class.getName(),
+                    String.class.getName(), boolean.class.getName(), String.class.getName(),  int.class.getName(),
+                    int.class.getName()};
+
+            Object result = mBeanServer.invoke(objectName,
+                    SubscriptionManagementConstants.FILTERED_SUBSCRIPTIONS_MBEAN_ATTRIBUTE,
+                    parameters, signature);
```
Comment (sdkottegoda) [inline components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/internal/registry/SubscriptionManagementBeans.java]: Can we shrink this to the previous line?

File: components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/internal/registry/SubscriptionManagementBeans.java [hunk 5]
```diff
@@ -92,6 +92,122 @@ public static SubscriptionManagementBeans getInstance(){
...
+    public int getTotalSubscriptionCountForSearchResult(boolean isDurable, boolean isActive, String protocolType,
+                                                        String destinationType, String filteredNamePattern,
+                                                        boolean isFilteredNameByExactMatch, String identifierPattern,
+                                                        boolean isIdentifierPatternByExactMatch, String ownNodeId)
+                                                        throws SubscriptionManagerException {
+        int totalSubscriptionCount = 0;
+        MBeanServer mBeanServer = ManagementFactory.getPlatformMBeanServer();
+        try {
+            ObjectName objectName =
+                    new ObjectName("org.wso2.andes:type=SubscriptionManagementInformation," +
```
Comment (sdkottegoda) [inline components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/internal/registry/SubscriptionManagementBeans.java]: + needs to be at the beginning

File: components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/internal/registry/SubscriptionManagementBeans.java [hunk 6]
```diff
@@ -92,6 +92,122 @@ public static SubscriptionManagementBeans getInstance(){
...
+            Object result = mBeanServer.invoke(objectName,
+                    SubscriptionManagementConstants.FILTERED_SUBSCRIPTION_COUNT_MBEAN_ATTRIBUTE,
+                    parameters, signature);
+
+            if (null != result) {
+                totalSubscriptionCount = (int) result;
+            }
+
+        } catch (MalformedObjectNameException | InstanceNotFoundException | MBeanException | ReflectionException e) {
+            throw new SubscriptionManagerException("Error while invoking mBean operations to get " +
```
Comment (sdkottegoda) [inline components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/internal/registry/SubscriptionManagementBeans.java]: + needs to be at the beginning

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp [hunk 7]
```diff
@@ -78,28 +83,94 @@
...
+        String filteredNameByExactMatch = request.getParameter("isQueueExactlyMatch");
+        String identifierPatternByExactMatch = request.getParameter("isIdentifierExactlyMatch");
+        boolean isFilteredNameByExactMatch = false;
+        boolean isIdentifierPatternByExactMatch = false;
+
+        if(null != filteredNameByExactMatch){
+            isFilteredNameByExactMatch = true;
+        }
+
+        if(null != identifierPatternByExactMatch){
```
Comment (sdkottegoda) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp]: Need space between 'if and (' and ') and {'

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp [hunk 8]
```diff
@@ -78,28 +83,94 @@
...
+            isIdentifierPatternByExactMatch = true;
+        }
+
+        if(filteredName == null || filteredName.trim().length() == 0){
```
Comment (sdkottegoda) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp]: Same as above

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp [hunk 9]
```diff
@@ -78,28 +83,94 @@
...
+            filteredName = "";
+        }
+        if(identifierPattern == null || identifierPattern.trim().length() == 0){
```
Comment (sdkottegoda) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp]: Same as above

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp [hunk 10]
```diff
@@ -78,28 +83,94 @@
...
+            %>
+            <script type="text/javascript">
+                location.href = "../admin/error.jsp";
+                alert("error");
+            </script>
+            <%
+                return;
+            }
+        String ownNodeId = request.getParameter("ownNodeId");
+        if(ownNodeId == null || ownNodeId.trim().length() == 0){
```
Comment (sdkottegoda) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp]: Same as above

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp [hunk 11]
```diff
@@ -78,28 +83,94 @@
...
+        Subscription[] filteredSubscriptionListForSearch;
         int subscriptionCountPerPage = 20;
         int pageNumber = 0;
         int numberOfPages = 1;
         String myNodeID;
-        String concatenatedParams = "region=region1&item=Queue_subscriptions";
+        String concatenatedParams = "region=region1&item=Queue_subscriptions&queueNamePattern="+ filteredName
+        + "&identifier=" + identifierPattern + "&ownNodeId=" + ownNodeId;
+
+        if(isFilteredNameByExactMatch){
```
Comment (sdkottegoda) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp]: Same as above

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp [hunk 12]
```diff
@@ -78,28 +83,94 @@
...
+            concatenatedParams += "&isQueueExactlyMatch="+ filteredNameByExactMatch;
+        }
+
+        if(isIdentifierPatternByExactMatch){
```
Comment (sdkottegoda) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp]: Same as above

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp [hunk 13]
```diff
@@ -78,28 +83,94 @@
...
             if (pageNumberAsStr != null) {
                 pageNumber = Integer.parseInt(pageNumberAsStr);
             }
-
-            if (subscriptionList != null) {
-                totalQueueSubscriptionCount = subscriptionList.length;
+            filteredSubscriptionList = stub.getFilteredSubscriptions(true, true, ProtocolType.AMQP.name(),
+            DestinationType.QUEUE.name(), filteredName, isFilteredNameByExactMatch, identifierPattern,
+            isIdentifierPatternByExactMatch, ownNodeId, pageNumber, subscriptionCountPerPage);
+            if (filteredSubscriptionList != null){
```
Comment (sdkottegoda) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp]: Shall we flip the null check

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp [hunk 14]
```diff
@@ -119,10 +190,89 @@
...
+                        </tr>
+
+                    </tbody>
+                </table>
+            </form>
+            <p>&nbsp;</p>
+
             <%
-                if (subscriptionList == null) {
+                if (filteredSubscriptionList == null) {
```
Comment (sdkottegoda) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp]: shall we flip the null check

File: components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/SubscriptionManagerService.java [hunk 15]
```diff
@@ -41,6 +41,52 @@
...
+	 * @param isIdentifierPatternByExactMatch  exactly match the identifier or not
+	 * @param ownNodeId node Id of the node which own the subscriptions
+	 * @param pageNumber  page number in the pagination table
+	 * @param subscriptionCountPerPage number of subscriptions to be shown in the UI per page
+	 * @return list of subscriptions matching to the search criteria
+	 * @throws SubscriptionManagerException throws when an error occurs
+	 */
+	public List<Subscription> getFilteredSubscriptions(boolean isDurable, boolean isActive, String protocolType,
+													   String destinationType, String filteredNamePattern,
+													   boolean isFilteredNameByExactMatch, String identifierPattern,
```
Comment (sajinidesilva) [inline components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/SubscriptionManagerService.java]: Incorrect formatting

File: components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/SubscriptionManagerServiceImpl.java [hunk 16]
```diff
@@ -48,6 +48,42 @@
     }
 
 	/**
+	 * {@inheritDoc}
+	 */
+	public List<Subscription> getFilteredSubscriptions(boolean isDurable, boolean isActive, String protocolType,
+													   String destinationType, String filteredNamePattern,
+													   boolean isFilteredNameByExactMatch, String identifierPattern,
+													   boolean isIdentifierPatternByExactMatch, String ownNodeId,
```
Comment (sajinidesilva) [inline components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/SubscriptionManagerServiceImpl.java]: Incorrect formatting

File: components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/internal/registry/SubscriptionManagementBeans.java (hunk 3 above)
Comment (sajinidesilva) [inline components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/internal/registry/SubscriptionManagementBeans.java]: Where there are plus sign and when you break the line the + sign should go the the second line

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/topic_subscriptions_list.jsp [hunk 17]
```diff
@@ -130,75 +135,98 @@
...
 <%
+
+    String filteredName = request.getParameter("topicNamePattern");
+    String identifierPattern = request.getParameter("identifier");
+    String filteredNameByExactMatch = request.getParameter("isTopicExactlyMatch");
+    String identifierPatternByExactMatch = request.getParameter("isIdentifierExactlyMatch");
+    boolean isFilteredNameByExactMatch = false;
+    boolean isIdentifierPatternByExactMatch = false;
+
+    if(null != filteredNameByExactMatch){
```
Comment (sdkottegoda) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/topic_subscriptions_list.jsp]: Incorrect formatting in all the 'if's

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp [hunk 18]
```diff
@@ -78,28 +83,94 @@
...
+        (CarbonConstants.CONFIGURATION_CONTEXT);
+        String cookie = (String) session.getAttribute(ServerConstants.ADMIN_SERVICE_COOKIE);
+        String nodeId = "";
+            try {
+                client = new ClusterManagerClient(configContext, serverURL, cookie);
+                isClusteringEnabled = client.isClusteringEnabled();
+                allClusterNodeAddresses = client.getAllClusterNodeAddresses();
+                nodeId = client.getMyNodeID();
+            } catch (Exception e) {
+                CarbonUIMessage.sendCarbonUIMessage(e.getMessage(), CarbonUIMessage.ERROR, request, e);
```
Comment (sajinidesilva) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp]: Wrong way of handling errors here. Need to remove line 121 and change the script

File: components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/internal/registry/SubscriptionManagementBeans.java (hunk 3 above)
Comment (sirojan24) [inline components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/internal/registry/SubscriptionManagementBeans.java]: Done in https://github.com/wso2/carbon-business-messaging/pull/357

File: components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/internal/registry/SubscriptionManagementBeans.java (hunk 4 above)
Comment (sirojan24) [inline components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/internal/registry/SubscriptionManagementBeans.java]: Done in https://github.com/wso2/carbon-business-messaging/pull/357

File: components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/internal/registry/SubscriptionManagementBeans.java (hunk 5 above)
Comment (sirojan24) [inline components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/internal/registry/SubscriptionManagementBeans.java]: Done in https://github.com/wso2/carbon-business-messaging/pull/357

File: components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/internal/registry/SubscriptionManagementBeans.java (hunk 6 above)
Comment (sirojan24) [inline components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/internal/registry/SubscriptionManagementBeans.java]: Fixed in https://github.com/wso2/carbon-business-messaging/pull/357

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp (hunk 7 above)
Comment (sirojan24) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp]: Corrected in https://github.com/wso2/carbon-business-messaging/pull/357

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp (hunk 8 above)
Comment (sirojan24) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp]: Corrected in https://github.com/wso2/carbon-business-messaging/pull/357

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp (hunk 9 above)
Comment (sirojan24) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp]: Corrected in https://github.com/wso2/carbon-business-messaging/pull/357

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp (hunk 10 above)
Comment (sirojan24) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp]: Corrected in https://github.com/wso2/carbon-business-messaging/pull/357

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp (hunk 11 above)
Comment (sirojan24) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp]: corrected in https://github.com/wso2/carbon-business-messaging/pull/357

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp (hunk 12 above)
Comment (sirojan24) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp]: Corrected in https://github.com/wso2/carbon-business-messaging/pull/357

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp (hunk 13 above)
Comment (sirojan24) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp]: flipped in https://github.com/wso2/carbon-business-messaging/pull/357

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp (hunk 14 above)
Comment (sirojan24) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp]: flipped in https://github.com/wso2/carbon-business-messaging/pull/357

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/topic_subscriptions_list.jsp (hunk 17 above)
Comment (sirojan24) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/topic_subscriptions_list.jsp]: Corrected in https://github.com/wso2/carbon-business-messaging/pull/357

File: components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/SubscriptionManagerService.java (hunk 15 above)
Comment (sirojan24) [inline components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/SubscriptionManagerService.java]: Corrected in https://github.com/wso2/carbon-business-messaging/pull/357

File: components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/SubscriptionManagerServiceImpl.java (hunk 16 above)
Comment (sirojan24) [inline components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/SubscriptionManagerServiceImpl.java]: Corrected in https://github.com/wso2/carbon-business-messaging/pull/357

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp (hunk 18 above)
Comment (sirojan24) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp]: Corrected in https://github.com/wso2/carbon-business-messaging/pull/357

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/topic_subscriptions_list.jsp (hunk 17 above)
Comment (sirojan24) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/topic_subscriptions_list.jsp]: Corrected in https://github.com/wso2/carbon-business-messaging/pull/357

Comment (sirojan24) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/topic_subscriptions_list.jsp]: Corrected in https://github.com/wso2/carbon-business-messaging/pull/357

Comment (sirojan24) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/topic_subscriptions_list.jsp]: Corrected in https://github.com/wso2/carbon-business-messaging/pull/357

Comment (sirojan24) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/topic_subscriptions_list.jsp]: Corrected in https://github.com/wso2/carbon-business-messaging/pull/357

File: components/andes/org.wso2.carbon.andes.admin/src/main/java/org/wso2/carbon/andes/admin/AndesAdminService.java [hunk 19]
```diff
@@ -466,6 +468,104 @@ public void closeSubscription(String subscriptionID, String destination, String
...
+        try {
+            SubscriptionManagerService subscriptionManagerService =
+                    AndesBrokerManagerAdminServiceDSHolder.getInstance().getSubscriptionManagerService();
+            subscriptionCountForSearchResult = subscriptionManagerService
+                    .getTotalSubscriptionCountForSearchResult(isDurable, isActive, protocolType, destinationType,
+                            filteredNamePattern, isFilteredNameByExactMatch, identifierPattern,
+                            isIdentifierPatternByExactMatch,ownNodeId);
+
+        } catch (SubscriptionManagerException e) {
+            String errorMessage = e.getMessage();
```
Comment (chathurace) [inline components/andes/org.wso2.carbon.andes.admin/src/main/java/org/wso2/carbon/andes/admin/AndesAdminService.java]: better to provide more descriptive error message with context information.

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp [hunk 20]
```diff
@@ -119,10 +190,89 @@
...
+                        </tr>
+                    </thead>
+                    <tbody>
+                        <tr>
+                            <td class="leftCol-big" style="padding-right: 0 !important;">Queue name pattern
+                            </td>
+                            <td>
+                                <input type="text" name="queueNamePattern" value="<%=filteredName%>"/>
+                                <%
+                                    if(isFilteredNameByExactMatch){
```
Comment (cnapagoda) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp]: Code formatting issue

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp [hunk 21]
```diff
@@ -119,10 +190,89 @@
...
+                                %>
+                                     <input type="checkbox" name="isQueueExactlyMatch" checked/>Match entire word only
+                                <%
+                                    }else {
```
Comment (cnapagoda) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp]: Code formatting issue

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp [hunk 22]
```diff
@@ -119,10 +190,89 @@
...
+                        </tr>
+                        <tr>
+                            <td class="leftCol-big" style="padding-right: 0 !important;">Select connected node ID </td>
+                             <td><select id="ownNodeId" name="ownNodeId">
+                                <%
+                                    try {
+                                        if (isClusteringEnabled) {
+                                %>
+                                    <option selected="selected" value="<%=ownNodeId%>"><%=ownNodeId%></option>
+                                    <% for(int i = 0; i < allClusterNodeAddresses.length; i++){ %>
```
Comment (cnapagoda) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp]: Code formatting issue.

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp [hunk 23]
```diff
@@ -119,10 +190,89 @@
...
+                            </select>
+                            </td>
+                        </tr>
+                        <tr>
+                            <td class="leftCol-big" style="padding-right: 0 !important;">Identifier pattern
+                            </td>
+                            <td>
+                                <input type="text" name="identifier" value="<%=identifierPattern%>"/>
+                                 <%
+                                    if(isIdentifierPatternByExactMatch){
```
Comment (cnapagoda) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp]: Code formatting issue.

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp [hunk 24]
```diff
@@ -119,10 +190,89 @@
...
+                                %>
+                                     <input type="checkbox" name="isIdentifierExactlyMatch" checked/>Match entire word only
+                                <%
+                                    }else {
```
Comment (cnapagoda) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp]: Code formatting issue.

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp (hunk 20 above)
Comment (sirojan24) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp]: fixed in https://github.com/wso2/carbon-business-messaging/pull/366

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp (hunk 21 above)
Comment (sirojan24) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp]: fixed in https://github.com/wso2/carbon-business-messaging/pull/366

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp (hunk 23 above)
Comment (sirojan24) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/subscriptions/queue_subscriptions_list.jsp]: fixed in https://github.com/wso2/carbon-business-messaging/pull/366

**Pull request merged**